### PR TITLE
Issue #2 fix

### PR DESCRIPTION
**Pull request by burlakko**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -4,6 +4,19 @@
 const mailjet = require('node-mailjet');
 const { removeUndefined } = require('strapi-utils');
 
+const normalizeTo = to => {
```
Comment (ScottAgirs) [inline lib/index.js:7]: Please rename `to` to `recipients`

File: lib/index.js [hunk 2]
```diff
@@ -4,6 +4,19 @@
...
+  return Array.isArray(to) ?
+   to.map(i =>
```
Comment (ScottAgirs) [inline lib/index.js:9]: Please rename `i` to `recipient`

File: lib/index.js [hunk 3]
```diff
@@ -4,6 +4,19 @@
...
+     typeof i === 'string'
+      ? {Email: i}
+      : i) : [
+        {
+          Email: to || settings.defaultTo,
+          Name: toName || settings.defaultToName,
+        },
```
Comment (ScottAgirs) [inline lib/index.js:16]: Looks like we're currently not passing in the `toName` and `settings` to `normalizeTo()` - could you please add and ensure it works? :)

File: lib/index.js [hunk 4]
```diff
@@ -37,12 +50,7 @@ module.exports = {
                   Email: from || settings.defaultFrom,
                   Name: fromName || settings.defaultFromName,
                 },
-                To: [
-                  {
-                    Email: to || settings.defaultTo,
-                    Name: toName || settings.defaultToName,
-                  },
-                ],
+                To: normalizeTo(to),
```
Comment (ScottAgirs) [inline lib/index.js:53]: Think we might want to pass in the `toName` and `settings` here

**Review by ScottAgirs — CHANGES_REQUESTED** [review]
Minor patches and we're good to go!

Thanks a bunch for the PR!

PS. if need any help, feel free to reach out!

**Comment by burlakko** [comment]
All valid points. But while implementing I found the `normalizeTo` function looks inconsistent, e.g. `recipients` params may be an array or a string, settings.defaultTo and `recipientName` (`toName`) - only a string.
Also I've found out that the plugin's API is different from the native one. Native API has only `to` and `settings.defaultTo` params which may be just an email address string, an `{Email: ..., Name: ...}` object or an array of strings or objects. This actually looks very flexible and simple. Maybe we should stick with this approach here? What do you think?

**Comment by ScottAgirs** [comment]
@burlakko sadly, deprecating `defaultToName` and/or `toName` is not an option, please consider adding extra handler to ensure capability. If nothing comes to mind, lmk if nothing comes to mind, I'll tackle this over the next few days.

**Comment by ScottAgirs** [comment]
@burlakko thanks again for your input, this should be addressed with [Release 3.3.0](https://github.com/ijsto/strapi-provider-email-mailjet/releases/tag/3.3.0)

**Comment by burlakko** [comment]
Oh I actually planned to deal with it on monday, but you already got it done. Thanx!

**Comment by burlakko** [comment]
Though I see a potential problem here.

Line 37: `To = [{ Email: to, Name: toName || settings.defaultToName }];` will stick the `defaultToName` even if you're using the different `to`.

Consider following example
```
...
  settings: {
      defaultTo: "john.doe@ijs.to",
      defaultToName: "Johnny Bravodoe",
    }
...
strapi.plugins["email"].services.email.send({to: 'hanna@whatever.email', ...})
```
It will result into `[{ Email: 'hanna@whatever.email', Name: 'Johnny Bravodoe' }]` recipient, which is obviously not what expected.

**Pull request closed without merging**